bitcoin/src/crypto/sighash.rs

```diff
@@ -1547,10 +1555,16 @@ mod tests {
         let script = ScriptBuf::new();
         let cache = SighashCache::new(&tx);
 
-        let got = cache.legacy_signature_hash(1, &script, SIGHASH_SINGLE).expect("sighash");
+        let sighash_type = 3;
+        let got = cache.legacy_signature_hash(1, &script, sighash_type).expect("sighash");
         let want = LegacySighash::from_byte_array(UINT256_ONE);
+        assert_eq!(got, want);
 
-        assert_eq!(got, want)
+        // https://github.com/rust-bitcoin/rust-bitcoin/issues/4112
+        let sighash_type = 131;
+        let got = cache.legacy_signature_hash(1, &script, sighash_type).expect("sighash");
+        let want = LegacySighash::from_byte_array(UINT256_ONE);
+        assert_eq!(got, want);
```

Comment:
Not a big deal but we sometimes put tests in separate commits (always ordered after the fix) to make it easier to demonstrate that the test fails before the fix.

Reply:
Test commit separated as you wish :P